cupyx/scipy/fft/fft.py

```diff
@@ -0,0 +1,418 @@
+import cupy
+from cupy.cuda import cufft
+from cupy.fft.fft import (_fft, _default_fft_func, hfft as _hfft,
+                          ihfft as _ihfft)
+from cupy.fft.fft import fftshift, ifftshift, fftfreq, rfftfreq
+import numpy as np
+from numbers import Number
+
+
+__all__ = ['fft', 'ifft', 'fft2', 'ifft2', 'fftn', 'ifftn',
+           'rfft', 'irfft', 'rfft2', 'irfft2', 'rfftn', 'irfftn',
+           'hfft', 'ihfft',
+           'fftshift', 'ifftshift', 'fftfreq', 'rfftfreq']
+
+try:
+    import scipy.fft as _scipy_fft
+except ImportError:
+    class _DummyModule:
+        def __getattr__(self, name):
+            return None
+
+    _scipy_fft = _DummyModule()
+
+# Backend support for scipy.fft
+
+__ua_domain__ = 'numpy.scipy.fft'
+_implemented = {}
+
+
+def __ua_convert__(dispatchables, coerce):
+    if coerce:
+        try:
+            replaced = [
+                cupy.asarray(d.value) if d.coercible and d.type == np.ndarray
+                else d.value for d in dispatchables]
+        except TypeError:
+            return NotImplemented
+    else:
+        replaced = [d.value for d in dispatchables]
+
+    if not all(d.type != np.ndarray or isinstance(r, cupy.ndarray)
+               for r, d in zip(replaced, dispatchables)):
+        return NotImplemented
+
+    return replaced
+
+
+def __ua_function__(method, args, kwargs):
+    fn = _implemented.get(method, None)
+    if fn is None:
+        return NotImplemented
+    return fn(*args, **kwargs)
+
+
+def _implements(scipy_func):
+    """Decorator adds function to the dictionary of implemented functions"""
+    def inner(func):
+        _implemented[scipy_func] = func
+        return func
+
+    return inner
+
+
+def _asiterable(x):
+    """Convert scalars to iterable, otherwise pass through ``x`` unchanged"""
```

Comment:
`_assequence` looks better because only `collections.abc.Sequence` instances are allowed for axes and shape.